sunpy/instr/rhessi.py

```diff
@@ -352,13 +350,20 @@ def uncompress_countrate(compressed_countrate):
     ----------
     Hsi_obs_summ_decompress.pro `<http://hesperia.gsfc.nasa.gov/ssw/hessi/idl/qlook_archive/hsi_obs_summ_decompress.pro>`_
     """
+
+    # Ensure uncompressed counts are between 0 and 255
+    if not ((compressed_countrate >= 0).all() and (compressed_countrate < 256).all()):
```

Comment:
Yes! I will update, it's a bit of a redundant check as the data that is passed to it is unit8 but if someone called with and int array or similar it will work but give nonsense results.